src/main/groovy/com/vherasymenko/avro/encoder/core/LessonStatusService.groovy

```diff
@@ -0,0 +1,75 @@
+package com.vherasymenko.avro.encoder.core
+
+import com.vherasymenko.avro.encoder.core.binary.AvroBinaryEncoderPort
+import com.vherasymenko.avro.encoder.outbound.MessageProducer
+import com.vherasymenko.avro.schared.AvroConstants
+import com.vherasymenko.avro.schared.MessageHeaders
+import groovy.json.JsonSlurper
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericData
+import org.apache.avro.generic.GenericRecord
+import org.springframework.messaging.support.MessageBuilder
+
+/**
+ * Service that encodes lesson status document. Also uses GenericRecord that works without code generation.
+ * This service uses nested schemas as well.
+ */
+class LessonStatusService implements LessonStatusPort {
+
+    /**
+     * The gateway to the messaging system.
+     */
+    private final MessageProducer producer
+
+    /**
+     * The binary encoder.
+     */
+    private final AvroBinaryEncoderPort encoder
+
+    LessonStatusService( MessageProducer aProducer, AvroBinaryEncoderPort anEncoder ) {
+        producer = aProducer
+        encoder = anEncoder
+    }
+
+    @Override
+    void handleEvent( String event ) {
+        def parser = new Schema.Parser()
+        def innerSchema = parser.parse( new File( 'src/main/avro/lessonStatusItem.avsc' ) )
```

Comment:
If you abstract away how the schema is loaded, then you can more easily test using embedded schemas vs using a registry.